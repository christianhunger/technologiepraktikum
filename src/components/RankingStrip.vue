<template>
  <ul class="rankingStrip">
    <li class="rankingStripOpponentWrapper" v-for="opponent in opponentsSortedByRating" :key="opponent.name">
      <img class="rankingStripOpponentImage" :src="require(`../assets/kitten${opponent.id}.jpg`)" />
    </li>
  </ul>
</template>

<script>
export default {
  name: "RankingStrip",
  props: {
    opponentsSortedByRating: Array
  }
};
</script>

<style scoped>
.rankingStripOpponentWrapper {
  display: inline-block;
}
.rankingStripOpponentImage {
  width: 64px;
  padding: 1px;
  transition: width 0.2s;
}
.rankingStripOpponentImage:hover {
  width: 100px;
  transition: width 0.1s;
}
</style>
