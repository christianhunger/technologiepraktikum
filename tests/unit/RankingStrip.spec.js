import { shallowMount } from "@vue/test-utils";
import RankingStrip from "@/components/RankingStrip";

describe("RankingStrip.vue", () => {
  const contendersSortedByRating = [
    { name: "Fluffykins", imageUrl: "fluffykins.png" },
    { name: "Furly", imageUrl: "furrrr.png" }
  ];

  it("renders a list of kittens when passed", () => {
    const wrapper = shallowMount(RankingStrip, {
      propsData: { contendersSortedByRating }
    });

    let html = wrapper.html();
    expect(html).toContain([contendersSortedByRating[0].imageUrl]);
    expect(html).toContain([contendersSortedByRating[1].imageUrl]);
    expect(wrapper.findAll("li")).toHaveLength(2);
  });

  it("sound match the snapshot", () => {
    const wrapper = shallowMount(RankingStrip, {
      propsData: { contendersSortedByRating }
    });

    expect(wrapper).toMatchSnapshot();
  });
});
