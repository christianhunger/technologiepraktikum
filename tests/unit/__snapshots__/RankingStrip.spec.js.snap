// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RankingStrip.vue sound match the snapshot 1`] = `
<ul class="rankingStrip">
  <li class="rankingStripContenderWrapper"><img src="fluffykins.png" class="rankingStripContenderImage"></li>
  <li class="rankingStripContenderWrapper"><img src="furrrr.png" class="rankingStripContenderImage"></li>
</ul>
`;
